### PR TITLE
feat: add support for information_schema.columns

### LINKED ISSUE #1390: Grafana compatibility: information_schema.columns table
### What problem does the new feature solve?

We have finished support for `information_schema.TABLES` in #1251, it's possible to list databases and tables in Grafana query builder. However, it further fetches columns information from another table of `information_schema`: `columns`.

The query grafana executes:

```sql
SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'public' AND table_name = 'host_cpu_seconds_total' ORDER BY column_name'
```

More information about `columns` table in mysql: https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-columns-table.html

### What does the feature do?

This feature allows our use to use MySQL query builder in Grafana when creating dashboards.

image

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fixes #1390 

This patch add a very basic `information_schema.columns` view support in order to fit grafana mysql plugin's query builder.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by sunng87** [comment]
Note that different from the `tables` view and mysql implementation, I don't include columns from `information_schema` itself for lack of mechanism to automate it. A manual implementation will be different to maintain.

I'm also thinking about removing `information_schema.tables` from its output as well, so that we hide tables/columns of `information_schema` from these queries totally. This is also consistent with our behaviour on `show databases`. WDTY @MichaelScofield

**Comment by MichaelScofield** [comment]
`information_schema` is rather customized, there's no "standard" of it. As long as it fits our needs, I'm OK with all the changes.

File: src/catalog/src/information_schema.rs [hunk 1]
```diff
@@ -50,31 +53,48 @@ impl SchemaProvider for InformationSchemaProvider {
...
-            })?,
-        );
         let table = TableAdapter::new(table).context(TableSchemaMismatchSnafu)?;
         Ok(Some(Arc::new(table)))
     }
 
     async fn table_exist(&self, name: &str) -> Result<bool> {
-        Ok(matches!(name.to_ascii_lowercase().as_str(), TABLES))
+        let normalized_name = name.to_ascii_lowercase();
+        Ok(normalized_name == TABLES || normalized_name == COLUMNS)
```
Comment (MichaelScofield) [inline src/catalog/src/information_schema.rs:98]:
```suggestion
        let normalized_name = name.to_ascii_lowercase();
        Ok(self.table_names()?.contains(&normalized_name))
```

**Review by MichaelScofield — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**